vagrant/provisioning/roles/elasticsearch-install/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+    # tasks file for /etc/ansible/roles/elastic
+    - name: Copy the ElasticSearch repo
+      template:
+        src: elasticsearch.repo
+        dest: /etc/yum.repos.d/
+        backup: yes
+    
+    - name: Import GPG key for ElasticSearch   
+      rpm_key:
+        state: present
+        key: https://artifacts.elastic.co/GPG-KEY-elasticsearch      
+
+    - name: Install ElasticSearch
+      yum:
+        name: elasticsearch
+        state: present
+
+    - name: Elasticsearch folders
+      become: yes
+      file:
+        path: "{{ item }}"
+        state: directory
+        group: elasticsearch
+        owner: elasticsearch
+      loop:
+        - "{{ root_folder }}/data/elasticsearch"
+        - "{{ root_folder }}/log/elasticsearch"
+
+    - name: Copying ElasticSearch configuration files
+      template:
+        src: elasticsearch.yml
+        dest: /etc/elasticsearch
+        backup: yes
+    
+    - name: Adding services to auto start
+      systemd:
+        name: elasticsearch
+        enabled: yes
+        daemon_reload: yes
+        masked: no
+        state: started
+      when: enable_elasticsearch is defined and enable_elasticsearch == "yes"
+ 
+    - name: Wait for elasticsearch to become responsive...
+      uri:
+        url: "{{ external_host | default('http://localhost') }}:{{ elasticsearch_port | default(9200) }}"
```

Comment:
For a single-VM install where httpd has already started using `external_host` probably works, but for AWS or other sites where the httpd/haproxy may not be in place yet, this might not work.  You might want to check localhost after all.

Also, consider including `validate_certs: no` to the parameters, in case the elastic host is on a self-signed cert.

Also, add the `when` from line 43 to this task as well, so you only wait for Elastic to start when elastic is enabled, and you don't wait for it to start when Elastic is disabled.